docs/3.0rc/develop/task-caching.mdx

```diff
@@ -0,0 +1,299 @@
+
+my_stateful_task() # sleeps again
+```
+
+## Customizing the cache
+
+Prefect allows users to configure task caching behavior in numerous ways.
+
+### Cache expiration
+
+All cache keys can optionally be given an _expiration_ through the `cache_expiration` keyword on 
+the [task decorator](https://prefect-python-sdk-docs.netlify.app/prefect/tasks/#prefect.tasks.task):
+This keyword accepts a `datetime.timedelta` specifying a duration for which the cached value should be
+considered valid.
+
+Providing an expiration value results in Prefect persisting an expiration timestamp alongside the result
+record for the task.
+This expiration is then applied to _all_ other tasks that may share this cache key.
+
+### Cache policies
+
+Cache policies can be composed and altered using basic Python syntax to form more complex policies.
+For example, all task policies except for `NONE` can be _added_ together to form new policies that combine
+the individual policies' logic into a larger cache key computation.
+Combining policies in this way therefore results in caches that are _easier_ to bust.
+
+For example:
+
+```python
+from prefect import task
+from prefect.cache_policies import TASK_SOURCE, INPUTS
+
+@task(cache_policy=TASK_SOURCE + INPUTS)
+def my_cached_task(x: int):
+    return x + 42
+```
+
+This task will rerun anytime you provide new values for `x`, _or_ anytime you change the underlying code.
+
+The `INPUTS` policy is a special policy that allows you to _subtract_ string values to signal ignoring
+certain task inputs:
+
+```python
+from prefect import task
+from prefect.cache_policies import INPUTS
+
+
+my_custom_policy = INPUTS - 'debug'
+
+@task(cache_policy=my_custom_policy)
+def my_cached_task(x: int, debug: bool = False):
+    print('running...')
+    return x + 42
+
+
+my_cached_task(1)
+my_cached_task(1, debug=True) # still uses the cache
+```
+
+### Cache key functions
+
+Users can configure custom cache policy logic through the use of cache key functions.
+A cache key function is a function that accepts two positional arguments:
+- The first argument corresponds to the `TaskRunContext`, which stores task run metadata. For example, 
+this object has attributes `task_run_id`, `flow_run_id`, and `task`, all of which can be used in your
+custom logic.
+- The second argument corresponds to a dictionary of input values to the task. For example, 
+if your task has the signature `fn(x, y, z)` then the dictionary will have keys "x", "y", and "z" with corresponding values that can be used to compute your cache key.
+
+This function can then be specified using the `cache_key_fn` argument on 
+the [task decorator](https://prefect-python-sdk-docs.netlify.app/prefect/tasks/#prefect.tasks.task).
+
+For example:
+
+```python
+def static_cache_key(context, parameters):
+    # return a constant
+    return "static cache key"
+
+
+@task(cache_key_fn=static_cache_key)
+def my_cached_task(x: int):
+    return x + 1
+```
+
+## Multi-task caching
```

Comment:
This is the only new section - the rest was cut/pasted directly from the write tasks doc